src/lib/Layout/PanelSideBarLayout/PanelSideBar/Context/PanelSideBarContext.tsx

```diff
@@ -87,6 +87,8 @@ export const PanelSideBarProvider = <TPanelItemId extends string, TPanelItem>(
     );
   };
 
+  const isIconShownOnSidebarCollapse = menuItems.find((x) => x.id === activePanelId)?.onSidebarCollapseOptions?.showIcon || false;
```

Comment:
Does it make sense to have this in the context and then exposed ? :) 
I have would suggest to have it passed from outside so that you can manually enable the feature, and then activate only for items with the showIcon prop.. what do you think?

Reply:
Not exposed anymore and passed as props wherever is used